### PR TITLE
Introduce docker build files for moulin

### PR DESCRIPTION
**Pull request by arminn**
Partly reused docker configuration and script from:
https://github.com/aoscloud/meta-aos-vm/pull/56

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

### CODE REVIEW

**Comment by rshym** [comment]
I refrain from reviewing due to lack of knowledge regarding containers.
I have read `doc/docker.md` and have no comments or objections.

File: doc/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    x11proto-core-dev libx11-dev lib32z1-dev ccache libgl1-mesa-dev \
+    libxml2-utils xsltproc unzip bc ninja-build \
+    python3-pyelftools python3-crypto libncurses5 libssl-dev -y
+
+RUN pip3 install pycryptodomex
+
+RUN apt-get install --reinstall -y ca-certificates
+
+# Clean up APT when done.
+RUN apt-get clean && rm -rf /tmp/* /var/tmp/*
```
Comment (lorc) [inline doc/Dockerfile:26]: This makes very little sense, as each `RUN` command creates new "layer" (container image).  So you will clean contents of one layer, but all data will be preserved in the previous layers. This is why it is recommended to chain `RUN` commands: 

```
# Install necessery for build packages
RUN apt-get update && apt-get install -y \
    apt-utils cpio python python3 python3-pip python3-pexpect xz-utils debianutils iputils-ping \
    python3-jinja2 libegl1-mesa pylint3 xterm \
    vim locales devscripts debhelper \
    gawk wget diffstat texinfo chrpath socat libsdl1.2-dev \
    python-crypto checkpolicy python3-git python3-github \
    bzr pigz m4 lftp openjdk-8-jdk git-core rsync \
    gnupg flex bison gperf build-essential zip curl zlib1g-dev \
    gcc-multilib g++-multilib libc6-dev-i386 lib32ncurses5-dev \
    x11proto-core-dev libx11-dev lib32z1-dev ccache libgl1-mesa-dev \
    libxml2-utils xsltproc unzip bc ninja-build \
    python3-pyelftools python3-crypto libncurses5 libssl-dev -y && \
    pip3 install pycryptodomex && \
    apt-get install --reinstall -y ca-certificates && \
    apt-get clean && rm -rf /tmp/* /var/tmp/*
```

File: doc/run_docker.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+echo "WORKSPACE         : $WORKSPACE"
+echo "DOCKER_IMAGE_NAME : $DOCKER_IMAGE_NAME"
+
+docker run \
+	--network=host \
+	-v "$HOME"/.ssh:/home/builder/.ssh \
+	-v "$HOME"/.gitconfig:/home/builder/.gitconfig \
+	-v "${WORKSPACE}":/home/builder/workspace \
+	-ti --rm "$DOCKER_IMAGE_NAME"
```
Comment (lorc) [inline doc/run_docker.sh:62]: Missing line feed at EOF.

File: doc/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,60 @@
+FROM ubuntu:20.04
+
+# Fix for tzdata for docker image build
+ENV TZ=UTC
+RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+
+# Install necessery for build packages
```
Comment (lorc) [inline doc/Dockerfile:7]: s/necessery/necessary

File: doc/Dockerfile (hunk 1 above)
Comment (arminn) [reply doc/Dockerfile]: Done

File: doc/run_docker.sh (hunk 2 above)
Comment (arminn) [reply doc/run_docker.sh]: Done

File: doc/Dockerfile (hunk 3 above)
Comment (arminn) [reply doc/Dockerfile]: Done

**Comment by arminn** [comment]
@lorc Comments addressed. RUN usage decreased. Added moulin to the PATH

**Comment by lorc** [comment]
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Comment by dsemenets** [comment]
I guess those command should be added
RUN sudo pip3 install gitpython
RUN sudo apt install python-clang-4.0
RUN sudo pip3 install clang

**Comment by arminn** [comment]
@dsemenets this version is for new build system, so your comment may be for old

**Pull request merged**